File: re-wiser/src/main/java/org/seal/wiser/backend/Email.java
package org.seal.wiser.backend;

import lombok.*;

import java.time.OffsetDateTime;
import java.util.Collection;
import java.util.LinkedList;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Email {
    private long id;
    private String from;
    private String to;
    private String cc;
    private String bcc;
    private String subject;
    private String text;
    private OffsetDateTime receivedDateTime;
    private Collection<Attachment> attachments = new LinkedList<>();
}

File: emposter-web/src/main/frontend/src/model/Email.ts
import {Attachment} from "./Attachment";

class Email {
    id: number;
    from: string;
    to: string;
    cc: string;
    bcc: string;
    subject: string;
    text: string;
    receivedDateTime: string;
    attachments: Array<Attachment>;

    constructor(id: number,
                from: string,
                to: string,
                cc: string,
                bcc: string,
                subject: string,
                text: string,
                receivedDateTime: string,
                attachments: Array<Attachment>) {
        this.id = id;
        this.from = from;
        this.to = to;
        this.cc = cc;
        this.bcc = bcc;
        this.subject = subject;
        this.text = text;
        this.receivedDateTime = receivedDateTime;
        this.attachments = attachments;
    }
}

export {Email}